dftimewolf/lib/collectors/grr_hosts.py

```diff
@@ -103,6 +103,39 @@ def _GetClientBySelector(self, selector):
       self.ModuleError('Could not get client for {0:s}'.format(
           selector), critical=True)
 
+    def _SeenLastMonth(timestamp):
+      """ Take a UTC timestamp and check if it is in the last month.
+
+      Args:
+        timestamp (int): A timestamp in microseconds.
+
+      Returns:
+        boolean: True if the timestamp is in last month from now.
+      """
+      last_seen_datetime = datetime.datetime.utcfromtimestamp(
+          timestamp / 1000000)
+      diff_year_in_month = (datetime.datetime.utcnow().year
+          - last_seen_datetime.year) * 12
+      diff_month_in_month = (datetime.datetime.utcnow().month
+          - last_seen_datetime.month)
+      diff_month = diff_year_in_month + diff_month_in_month
+      return bool(diff_month < 1)
+
+    last_month_list = list(filter(lambda x: _SeenLastMonth(x[0]), result))
```

Comment:
I would rather move this logic to a separate function that takes a list of clients (`result`) and returns only the ones that were active in the past month (along with the logic in `_SeenLastMonth`)

If the list is empty, we should output an error message saying that no matching clients have been online for the past month. If the list is >1, then output the error message as below (also maybe stating how many clients exactly were returned by the query?)

Reply:
Done